src/arrays.jl

```diff
@@ -135,19 +138,23 @@ function Base.Broadcast.broadcasted(
     u::UniFinArr{S,V,R,P,N},
     cv::CategoricalValue) where {S,V,R,P,N}
 
-    cv in classes(u) || throw(err_missing_class(cv))
+    # we assume that we compare categorical values by their unwrapped value
+    # and pick the index of this value from classes(u)
+    cv_loc = findfirst(==(cv), classes(u))
+    cv_loc == 0 && throw(err_missing_class(cv))
 
     f() = zeros(P, size(u)) #default caller function
 
     return Base.Broadcast.Broadcasted(
         identity,
-        (get(f, u.prob_given_ref, int(cv)),)
+        (get(f, u.prob_given_ref, cv_loc),)
         )
 end
+
 Base.Broadcast.broadcasted(
     ::typeof(pdf),
     u::UniFinArr{S,V,R,P,N},
-    ::Missing) where {S,V,R,P,N} = Missings.missings(P, length(u))
+    ::Missing) where {S,V,R,P,N} = Missings.missings(P, size(u))
 
 # pdf.(u, v)
 function Base.Broadcast.broadcasted(
```

Comment:
As commented in the main thread I find this definition surprising, but maybe this is what you want (thought, the definition here is different than the one in Distributions.jl).

Reply:
@bkamins I agree. The definition doesn't quite behave as expected. The reason for this method is to take advantage of the structure of the `UnivariateFiniteArray` in the case where (`length.(axes(u)) == length.(axes(v))`). But for the example you gave in the main thread, this isn't the case. Hence the erroneous result.
```julia
julia> u = UnivariateFinite(['x', 'z'], rand(2, 3, 2), pool=missing, ordered=true)
2×3 UnivariateFiniteArray{OrderedFactor{2}, Char, UInt8, Float64, 2}:
 UnivariateFinite{OrderedFactor{2}}(x=>0.931, z=>0.78)   …  UnivariateFinite{OrderedFactor{2}}(x=>0.554, z=>0.222)
 UnivariateFinite{OrderedFactor{2}}(x=>0.997, z=>0.445)     UnivariateFinite{OrderedFactor{2}}(x=>0.401, z=>0.622)

julia> l = u[1:2, 1:1]
2×1 UnivariateFiniteArray{OrderedFactor{2}, Char, UInt8, Float64, 2}:
 UnivariateFinite{OrderedFactor{2}}(x=>0.931, z=>0.78)
 UnivariateFinite{OrderedFactor{2}}(x=>0.997, z=>0.445)

julia> c = permutedims(classes(l))
1×2 CategoricalArrays.CategoricalArray{Char,2,UInt8}:
 'x'  'z'

julia> Base.Broadcast.combine_axes(l, c)
(Base.OneTo(2), Base.OneTo(2))

julia> length.(axes(l)) == length.(axes(c))
false
```
I believe we need to add an `if ... else ... end` block in the current code, then just use the default fallback in the else case. (Even though this is slower).
I think we can do that in a separate PR.

Reply:
I agree. That is why I have not added it here as it is a separate issue. I only added tests so that we can later easily track it when things are fixed.